### ISSUE TITLE
Use Skidl + PCBNew for extensibility

### ISSUE THREAD

**Issue by cgarcia2097** [jeroen94704/klepcbgen#9]
Do you think it's possible to migrate the project from using text templates to using something like [Skidl](https://xess.com/skidl/docs/_site/index.html) for the netlist creation and PCBNew for the part placement? I think that expands the capability of the entire project without being limited to templates.

**Comment by jeroen94704** [comment]
Yes, I found out about SKidl (and I believe there are a few more) recently, and plan on using it in the future. Especially netlist creation is really tedious to get right with templates, so this would be much improved.

Thanks for the suggestion!

**Comment by cgarcia2097** [comment]
I have an alpha forked on my page that I quickly whipped up on a weekend.
It only does basic netlist generation and keyswitch placement.

If you want to take a look at it, that would be great.

**Comment by adamws** [comment]
There are few tools with similar functionality, one of them ([kicad-kbplacer](https://github.com/adamws/kicad-kbplacer)) is mine. It uses pcbnew python API and is available via kicad's plugin manager. I also integrated it with another tool which generates projects ([keyboard-tools](https://github.com/adamws/keyboard-tools)) and skidl is used there.
